src/sonymn_int.cpp

```diff
@@ -791,21 +792,87 @@ namespace Exiv2 {
         return tagInfoCs2_;
     }
 
+    //! Lookup table to translate Sony2Fp AF Area Mode values to readable labels
+    constexpr TagDetails sony2FpAFAreaMode[] = {
+        { 0,   N_("Multi")                               },
+        { 1,   N_("Center")                              },
+        { 2,   N_("Spot")                                },
+        { 3,   N_("Flexible Spot")                       },
+        { 10,  N_("Selective (for Miniature effect)")    },
+        { 11,  N_("Zone")                                },
+        { 12,  N_("Expanded Flexible Spot")              },
+        { 14,  N_("Tracking")                            },
+        { 15,  N_("Face Tracking")                       },
+        { 20,  N_("Animal Eye Tracking")                 },
+        { 255, N_("Manual")                              }
+    };
+
     //! Sony Tag 9402 Sony2Fp (FocusPosition)
     constexpr TagInfo SonyMakerNote::tagInfoFp_[] = {
-        {  0x04, "AmbientTemperature", N_("Ambient Temperature"), N_("Ambient Temperature"), sony2FpId, makerTags,   signedByte, 1, printValue},
-        {  0x16, "FocusMode"         , N_("Focus Mode")         , N_("Focus Mode")         , sony2FpId, makerTags, unsignedByte, 1, printValue},
-        {  0x17, "AFAreaMode"        , N_("AF Area Mode")       , N_("AF Area Mode")       , sony2FpId, makerTags, unsignedByte, 1, printValue},
-        {  0x2d, "FocusPosition2"    , N_("Focus Position 2")   , N_("Focus Position 2")   , sony2FpId, makerTags, unsignedByte, 1, printValue},
+        {  0x04, "AmbientTemperature", N_("Ambient temperature"), N_("Temperature of the surroundings (in degrees Celsius)"), sony2FpId, makerTags, signedByte, 1, printTemperatureInDegC},
+        {  0x16, "FocusMode"         , N_("Focus mode")         , N_("Focus mode")          , sony2FpId, makerTags, unsignedByte, 1, printSony2FpFocusMode},
+        {  0x17, "AFAreaMode"        , N_("AF area mode")       , N_("Auto focus area mode"), sony2FpId, makerTags, unsignedByte, 1, EXV_PRINT_TAG(sony2FpAFAreaMode)},
+        {  0x2d, "FocusPosition2"    , N_("Focus position 2")   , N_("Focus position 2")    , sony2FpId, makerTags, unsignedByte, 1, printSony2FpFocusPosition2},
         // End of list marker
-        {0xffff, "(Unknownsony2FpTag)", "(Unknownsony2FpTag)"   , "(Unknownsony2FpTag)"    , sony2FpId, makerTags, unsignedByte, 1, printValue},
+        {0xffff, "(UnknownSony2FpTag)", "(UnknownSony2FpTag)"   , "(UnknownSony2FpTag)"     , sony2FpId, makerTags, unsignedByte, 1, printValue},
     };
 
     const TagInfo* SonyMakerNote::tagListFp()
     {
         return tagInfoFp_;
     }
 
+    std::ostream& SonyMakerNote::printSony2FpFocusMode(std::ostream& os, const Value& value, const ExifData*)
+    {
+        if (value.count() != 1)
+            os << value;
+        else {
+            long val = (value.toLong() & 0x7F);
+            if (val == 0)
```

Comment:
What do you think about turning this into a switch statement? 
That's usually preferred over `if () else if() ...`, see e.g. [CppCoreGuidelines ES.70](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-switch-if)

Reply:
I changed these over. I have other tags still to submit and will use this in future.